Classes/Domain/Service/NodePropertyConversionService.php

```diff
@@ -113,6 +111,35 @@ public function convert(NodeType $nodeType, string $propertyName, string|array|n
         }
     }
 
+    /**
+     * @param array<int|string, mixed> $data
+     */
+    public function convertNodeCreationElements(NodeType $nodeType, array $data): NodeCreationElements
+    {
+        $convertedElements = [];
+        /** @var string $elementName */
+        foreach ($nodeType->getConfiguration('ui.creationDialog.elements') ?? [] as $elementName => $elementConfiguration) {
+            $rawValue = $data[$elementName] ?? null;
+            if ($rawValue === null) {
+                continue;
+            }
+            $propertyType = $elementConfiguration['type'] ?? 'string';
+            if ($propertyType === 'references' || $propertyType === 'reference') {
+                $nodeAggregateIds = [];
+                if (is_string($rawValue) && !empty($rawValue)) {
+                    $nodeAggregateIds = [$rawValue];
+                } elseif (is_array($rawValue)) {
+                    $nodeAggregateIds = $rawValue;
+                }
+                $convertedElements[$elementName] = NodeAggregateIds::fromArray($nodeAggregateIds);
```

Comment:
```suggestion
                $convertedElements[$elementName] = $this->convertReferenceOrReferences($rawValue);
```

Just a refactoring nitpick :innocent: